mcp/src/databricks_mcp/mcp.py

```diff
@@ -0,0 +1,130 @@
+        path = urlparse(self.mcp_url).path
+        return any(re.match(pattern, path) for pattern in MCP_URL_PATTERNS.values())
+
+    def _get_databricks_managed_mcp_url_type(self) -> str:
+        """Determine the MCP URL type based on the path."""
+        path = urlparse(self.mcp_url).path
+        for mcp_type, pattern in MCP_URL_PATTERNS.items():
+            if re.match(pattern, path):
+                return mcp_type
+        raise ValueError(f"Unrecognized MCP URL: {self.mcp_url}")
+
+    async def _get_tools_async(self) -> List[Tool]:
+        """Fetch tools from the MCP endpoint asynchronously."""
+        async with streamablehttp_client(
+            url=self.mcp_url,
+            auth=DatabricksOAuthClientProvider(self.client),
+        ) as (read_stream, write_stream, _):
+            async with ClientSession(read_stream, write_stream) as session:
+                await session.initialize()
+                return (await session.list_tools()).tools
+
+    async def _call_tools_async(
+        self,
+        tool_name: str,
+        arguments: dict[str, Any] | None = None,
+    ) -> CallToolResult:
+        """Call the tool with the given name and input."""
+        async with streamablehttp_client(
+            url=self.mcp_url,
+            auth=DatabricksOAuthClientProvider(self.client),
+        ) as (read_stream, write_stream, _):
+            async with ClientSession(read_stream, write_stream) as session:
+                return await session.call_tool(tool_name, arguments)
+
+    def _extract_genie_id(self) -> str:
+        """Extract the Genie space ID from the URL."""
+        path = urlparse(self.mcp_url).path
+        if "/genie/" not in path:
+            raise ValueError(f"Missing /genie/ segment in: {self.mcp_url}")
+        genie_id = path.split("/genie/", 1)[1]
+        if not genie_id:
+            raise ValueError(f"Genie ID not found in: {self.mcp_url}")
+        return genie_id
+
+    def _normalize_tool_name(self, name: str) -> str:
+        """Convert double underscores to dots for compatibility."""
+        return name.replace("__", ".")
+
+    def list_tools(self) -> List[Tool]:
+        """List the tools from the MCP URL."""
+        return asyncio.run(self._get_tools_async())
+
+    def call_tool(self, tool_name: str, arguments: dict[str, Any] | None = None) -> CallToolResult:
+        """Call the tool with the given name and input."""
+        return asyncio.run(self._call_tools_async(tool_name, arguments))
+
+    def get_databricks_resources(self) -> List[DatabricksResource]:
+        """
+        This method is used to get the Databricks resources from the MCP URL.
+        It will return a list of Databricks resources.
+
+        Note that this method is only for databricks managed MCP URLs. It returns empty list for other MCP URLs
+        """
+        try:
+            if not self._is_valid_databricks_managed_mcp_url():
+                logger.warning(f"Ignoring invalid MCP URL: {self.mcp_url}")
```

Comment:
```suggestion
                logger.warning(f"MCP server URL {self.server_url} did not match any known Databricks-managed MCP server URL. Returning an empty list of resources")
```

Reply:
Also, it's probably simpler if we just have an API like `_get_databricks_managed_mcp_url_type` which returns `None` if the type doesn't match any known ones, and then we handle that in the caller (instead of first doing this `_is_valid_databricks_managed_mcp_url` check and then calling `_get_databricks_managed_mcp_url_type`, needing to keep logic in sync across the two methods, etc)